src/plugins/discover/public/application/utils/state_management/discover_slice.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Filter, Query } from '../../../../../data/public';
+import { DiscoverServices } from '../../../build_services';
+import { RootState } from '../../../../../data_explorer/public';
+
+export interface DiscoverState {
+  /**
+   * Columns displayed in the table
+   */
+  columns?: string[];
+  /**
+   * Array of applied filters
+   */
+  filters?: Filter[];
+  /**
+   * Used interval of the histogram
+   */
+  interval?: string;
+  /**
+   * Lucence or DQL query
+   */
+  query?: Query;
+  /**
+   * Array of the used sorting [[field,direction],...]
+   */
+  sort?: string[][];
+  /**
+   * id of the used saved query
+   */
+  savedQuery?: string;
+}
+
+export interface DiscoverRootState extends RootState {
+  discover: DiscoverState;
+}
+
+const initialState = {} as DiscoverState;
+
+export const getPreloadedState = async ({ data }: DiscoverServices): Promise<DiscoverState> => {
+  return {
+    ...initialState,
+    interval: data.query.timefilter.timefilter.getRefreshInterval().value.toString(),
```

Comment:
Why is the interval the only part retrieved from the data plugin?

Reply:
This is just to validate that the state management is working as expected across views. Its placeholder content.